services/auth/source/ldap/source_search.go

```diff
@@ -117,10 +117,11 @@ func dial(source *Source) (*ldap.Conn, error) {
 	}
 
 	if source.SecurityProtocol == SecurityProtocolLDAPS {
-		return ldap.DialTLS("tcp", net.JoinHostPort(source.Host, strconv.Itoa(source.Port)), tlsConfig)
+		dialOpt := ldap.DialWithTLSConfig(tlsConfig)
+		return ldap.DialURL(net.JoinHostPort(source.Host, strconv.Itoa(source.Port)), dialOpt)
 	}
 
-	conn, err := ldap.Dial("tcp", net.JoinHostPort(source.Host, strconv.Itoa(source.Port)))
+	conn, err := ldap.DialURL(net.JoinHostPort(source.Host, strconv.Itoa(source.Port)))
```

Comment:
`ldap://` here

Reply:
I added it in, although I think by adding it in, it may alter the behaviour. As looking at the diff for the ldap import changes, previously it handled the missing protocol the same way as it does now.

Reply:
What triggered this change? A deprecation? `DialURL` sounds like it should only be passed URLs, but I guess better to look at the source code of those functions because the docs are lacking.

Reply:
Here's the diff: https://github.com/go-ldap/ldap/compare/v3.4.6...v3.4.8#diff-4f427d2b022907c552328e63f137561f6de92396d7a6e8f6c2ea1bcf0db52654

Looks like those functions were already deprecated, it's just a change in comments allowed it to be picked up by our linter, and that the functions we used didn't modify the protocol at all. So I will remove my additions of hardcoding a protocol.

Reply:
I think it's wrong to exclude the scheme. A URL without a scheme is not a URL.

Have a look at https://github.com/go-ldap/ldap/blob/06d50d1ad03bcd323e48f2fe174d95ceb31b8b90/conn.go#L229-L251 and specifically https://github.com/go-ldap/ldap/blob/06d50d1ad03bcd323e48f2fe174d95ceb31b8b90/conn.go#L248C10-L248C17, it shows TLS is only enable when `u.Scheme === "ldap"` which can not be true with the current argument passed to `DialURL`.

Reply:
fair. I'm not sure I can run through a full testing of LDAP right now, so what if I revert the ldap changes, and revisit it in an upcoming PR?

Reply:
Seems like a harmless change to do imho. Actual deprecation came in https://github.com/go-ldap/ldap/commit/2517798aa0227380815f7a7106a194d4c7ca3bf9 and in their test suite, you can also see the change from host:port to scheme://host:port syntax.

Reply:
If you really must, add a `//nolint` for it, but I'd much prefer to do the change and run the ldap tests which should be running as part of CI anyways.